docs/src/main/asciidoc/mongodb-panache-guide.adoc

```diff
@@ -0,0 +1,525 @@
+
+Here are some query examples:
+
+- `firstname = ?1 and status = ?2` will be mapped to `{'firstname': ?1, 'status': ?2}`
+- `amount > ?1 and firstname != ?2` will be mapped to `{'amount': {'$gt': ?1}, 'firstname': {'$ne': ?2}}`
+- `lastname like ?1` will be mapped to `{'lastname': {'$regex': ?1}}`. Be careful that this will be link:https://docs.mongodb.com/manual/reference/operator/query/regex/#op._S_regex[MongoDB regex] support and not SQL like pattern. 
+- `lastname is not null` will be mapped to `{'lastname':{'$exists': true}}`
+
+We also handle some basic date type transformations: all fields of type `Date`, `LocalDate` or `LocalDateTime` will be mapped to the link:https://docs.mongodb.com/manual/reference/bson-types/#document-bson-type-date[BSON Date] using the `ISODate` type (UTC datetime).
+
+MongoDB with Panache also supports extended MongoDB queries by providing a `Document` query, this is supported by the find/list/stream/cout/delete methods.
+
+== Query parameters
+
+You can pass query parameters, for both native and PanacheQL queries, by index (1-based) as shown below:
+
+[source,java]
+--
+Person.find("name = ?1 and status = ?2", "Loïc", Status.Alive);
+Person.find("{'name': ?1, 'status': ?2}", "Loïc", Status.Alive);
+--
+
+Or by name using a `Map`:
+
+[source,java]
+--
+Map<String, Object> params = new HashMap<>();
+params.put("name", "Loïc");
+params.put("status", Status.Alive);
+Person.find("name = :name and status = :status", params);
+Person.find("{'name': :name, 'status', :status}", params);
+--
+
+Or using the convenience class `Parameters` to either build a `Map` or just used as-is:
+
+[source,java]
+--
+// generate a Map
+Person.find("name = :name and status = :status",
+         Parameters.with("name", "Loïc").and("status", Status.Alive).map());
+
+// use it as-is
+Person.find("{'name': :name, 'status': :status}",
+         Parameters.with("name", "Loïc").and("status", Status.Alive));
+--
+
+Every query operation accepts passing parameters by index (`Object...`), or by name (`Map<String,Object>` or `Parameters`).
+
+When you use query parameters, be careful that PanacheQL queries will refers to the Object parameters name but native queries will refers to MongoDB field names.
+
+Imagine the following entity:
+
+[source,java]
+--
+public class Person extends PanacheMongoEntity {
+    @BsonProperty("lastname")
+    public String name;
+    public LocalDate birth;
+    public Status status;
+
+    public static Person findByNameWithPanacheQLQuery(String name){
+      return find("name", name).firstResult();
+    }
+
+    public static Person findByNameWithNameQuery(String name){
```

Comment:
You call them NativeQuery until now. Let's be consistent? (`s/NameQuery/NativeQuery/`)

Reply:
typo, corrected